sympy/core/expr.py

```diff
@@ -781,7 +781,11 @@ def _eval_interval(self, x, a, b):
         else:
             A = self.subs(x, a)
             if A.has(S.NaN, S.Infinity, S.NegativeInfinity, S.ComplexInfinity):
-                A = limit(self, x, a)
+                if a.__lt__(b):
```

Comment:
Yes, the typical way is `if (a < b) == True` (be careful, don't use `is`, since it gives `sympy.true`, not `True`). The issue is that if you just use `if a < b` then if the inequality is symbolic (can't be evaluated, like if a and b are symbols), then it will raise a `TypeError`.